src/elements/datepicker/common/datepicker-association-controllers.ts

```diff
@@ -0,0 +1,115 @@
+import type { LitElement, ReactiveController } from 'lit';
+
+import type { SbbDatepickerElement } from '../datepicker/datepicker.js';
+
+export interface SbbDatepickerControl<T> extends LitElement {
+  datepicker: SbbDatepickerElement<T> | null;
+}
+
+class SbbDatepickerAssociationContext<T> {
+  private static _registry = new WeakMap<Node, SbbDatepickerAssociationContext<any>>();
+
+  public readonly hosts = new Set<SbbDatepickerElement<T>>();
```

Comment:
for me it would be better encapsulated to not expose hosts and controls, but have specific "add" and "find" methods to interact with it, like deleteControls(). Not very important :)

Reply:
I thought about that, but the controllers need access to the set and abstracting it with a getter felt overkill for a non-public API.

Reply:
it's ok like it is, but I meant not just getters but specific methods which interact with it (like filtering, or whatever manipulation)